sdk/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -163,81 +192,56 @@ public void forceFlush() {
               Labels.of("spanProcessorType", BatchSpanProcessor.class.getSimpleName()));
     }
 
-    private static final BoundLongCounter droppedSpans;
-
     private final ExecutorService executorService =
         Executors.newSingleThreadExecutor(new DaemonThreadFactory(EXPORTER_THREAD_NAME));
-
-    private static final Logger logger = Logger.getLogger(Worker.class.getName());
     private final SpanExporter spanExporter;
-    private final long scheduleDelayMillis;
     private final int maxQueueSize;
     private final int maxExportBatchSize;
     private final int halfMaxQueueSize;
-    private final Object monitor = new Object();
     private final int exporterTimeoutMillis;
-
-    @GuardedBy("monitor")
-    private final List<ReadableSpan> spansList;
+    private final BlockingQueue<ReadableSpan> spansList;
 
     private Worker(
         SpanExporter spanExporter,
-        long scheduleDelayMillis,
         int maxQueueSize,
         int maxExportBatchSize,
         int exporterTimeoutMillis) {
       this.spanExporter = spanExporter;
-      this.scheduleDelayMillis = scheduleDelayMillis;
       this.maxQueueSize = maxQueueSize;
       this.halfMaxQueueSize = maxQueueSize >> 1;
       this.maxExportBatchSize = maxExportBatchSize;
-      this.spansList = new ArrayList<>(maxQueueSize);
+      this.spansList = new ArrayBlockingQueue<>(maxQueueSize);
       this.exporterTimeoutMillis = exporterTimeoutMillis;
     }
 
+    private static List<SpanData> createSpanDataForExport(
+        List<ReadableSpan> spanList, int startIndex, int numberToTake) {
+      List<SpanData> spanDataBuffer = new ArrayList<>(numberToTake);
```

Comment:
Due to th edeceptive name `numberToTake`, this looks like a bug, the list gets bigger and bigger and bigger :)